### PR TITLE
improve networking: Allow to set options in resolv.conf.

### PR DESCRIPTION
**Pull request by zagy**
This allows to set all options in /etc/resolv.conf.

### CODE REVIEW

**Comment by zimbatm** [comment]
nice. why hasn't this been merged yet ?

**Comment by joachifm** [comment]
I want this, if it works as advertised. I prefer to interpret lack of feedback as silent assent, and would merge this if rebased.

**Comment by zagy** [comment]
Rebased.

File: nixos/modules/config/networking.nix [hunk 1]
```diff
@@ -59,6 +64,14 @@ in
       '';
     };
 
+    networking.resolvconfOptions = lib.mkOption {
+      type = types.lines;
```
Comment (peti) [inline nixos/modules/config/networking.nix]: Would a list of strings not would be a better type for this kind of information?

Comment (joachifm) [inline nixos/modules/config/networking.nix]: You're right, seems like it would simplify the processing of the option value, too.

**Comment by joachifm** [comment]
I have implemented @peti's suggestion at https://github.com/joachifm/nixpkgs/commits/pr-11372; I'm a little unsure whether
```nix
{
  networking.resolvconfOptions = [ "ndots:1" "rotate" ];
}
```
is really that much better than
```nix
{
  networking.extraResolvconfConf = ''
    resolv_conf_options+=' ndots:1 rotate'
  '';
}
```
though ... Thoughts?

**Comment by peti** [comment]
I think the first approach is superior because that list of options is easier to inspect within Nix than the second form, which is more or less opaque to further processing. Also, as a user I don't have to know the resolvconf syntax (with leading space and all) just to set options.

**Comment by joachifm** [comment]
Makes sense, I've pushed it.

**Pull request closed without merging**